homeassistant/components/zamg/sensor.py

```diff
@@ -242,13 +248,17 @@ class ZamgSensor(CoordinatorEntity, SensorEntity):
     entity_description: ZamgSensorEntityDescription
 
     def __init__(
-        self, coordinator, name, station_id, description: ZamgSensorEntityDescription
-    ):
+        self,
+        coordinator: ZamgDataUpdateCoordinator,
+        name: str,
+        station_id: str,
+        description: ZamgSensorEntityDescription,
+    ) -> None:
         """Initialize the sensor."""
         super().__init__(coordinator)
         self.entity_description = description
         self._attr_name = f"{name} {description.name}"
-        self._attr_unique_id = f"{station_id}_{description.key}"
+        self._attr_unique_id = f"{DOMAIN}_{station_id}_{description.key}"
```

Comment:
The entity registry is aware of both the integration domain the the platform domain. We don't need a domain prefix.

Reply:
Ok, should I remove `DOMAIN` here and in `weather.py`?

Reply:
Yes, please. And we should merge this asap so it's included in the release of today. Otherwise we need to migrate the entities in the entity registry to the new unique id.

Reply:
```suggestion
        self._attr_unique_id = f"{station_id}_{description.key}"
```